tracer/parser.go

```diff
@@ -52,6 +56,15 @@ func NewParser(global *global.Global, event *event.Event) *Parser {
 	}
 }
 
+func (p *Parser) UseLogs(l *logger.Logger, eventID uuid.UUID) {
+	logs, err := l.FinishRead(eventID)
+	if err != nil {
+		slog.Error("Failed to finish logs read", "eventID", eventID, "err", err)
+	} else {
+		p.logs = logs
```

Comment:
this is a data race.

the expectation is that `(*Parser).Finish()` can be called in a different goroutine from the `UseRequest` and `UseResponse` ones (and therefore `UseLogs` too). the other two `Use` functions have synchronisation in the form of a `<-p.errs`, but `UseLogs` doesn't.